include/sys/atomic.h

```diff
@@ -22,6 +22,7 @@ extern "C" {
 typedef int atomic_t;
 typedef atomic_t atomic_val_t;
 typedef void *atomic_ptr_t;
+typedef atomic_ptr_t atomic_ptr_val_t;
```

Comment:
why are we introducing `atomic_ptr_val_t` instead of `void *`?

Reply:
Following the pattern used for `atomic_t`/`atomic_val_t`.